src/formatDate.js

```diff
@@ -1,56 +1,68 @@
 'use strict';
 
 /**
- *   Time flies, standards change. Let's get rid of the routine of changing the
- * date format. Create a `formatDate` function that accepts the `date` string,
- * the old `fromFormat` array and the new `toFormat` array. Function returns
- * given date in new format.
- *   The function can change a separator, reorder the date parts of convert a
- * year from 4 digits to 2 digits and back.
- *   When converting from YYYY to YY just use 2 last digit (1997 -> 97).
- *   When converting from YY to YYYY use 20YY if YY < 30 and 19YY otherwise.
- *
- * Examples:
- *
- * formatDate(
- *   '2020-02-18',
- *   ['YYYY', 'MM', 'DD', '-'],
- *   ['YYYY', 'MM', 'DD', '.'],
- * ) // '2020.02.18'
- *
- * formatDate(
- *   '2020-02-18',
- *   ['YYYY', 'MM', 'DD', '-'],
- *   ['DD', 'MM', 'YYYY', '.'],
- * ) // '18.02.2020'
- *
- * formatDate(
- *   '18-02-2020',
- *   ['DD', 'MM', 'YYYY', '-'],
- *   ['DD', 'MM', 'YY', '/'],
- * ) // '18/02/20'
- *
- * formatDate(
- *   '20/02/18',
- *   ['YY', 'MM', 'DD', '/'],
- *   ['YYYY', 'MM', 'DD', '.'],
- * ) // '2020.02.18'
- *
- * formatDate(
- *   '97/02/18',
- *   ['YY', 'MM', 'DD', '/'],
- *   ['DD', 'MM', 'YYYY', '.'],
- * ) // '18.02.1997'
- *
  * @param {string} date
  * @param {string[]} fromFormat
  * @param {string[]} toFormat
  *
  * @returns {string}
  */
-
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  return arrangeDate(parseDate(date, fromFormat), toFormat);
+}
+
+/**
+ * @param {string} date
+ * @param {string[]} format
+ *
+ * @returns {Object}
+ */
+function parseDate(date, format) {
+  const YEAR_FORMAT_SHORT = 'YY';
+  const YEAR_FORMAT_LONG = 'YYYY';
+  const CENTURY_THRESHOLD = 30;
+  const CENTURY_PREFIX_20 = 19;
+  const CENTURY_PREFIX_21 = 20;
+
+  const parts = format.slice(0, -1);
+  const separator = format.slice(-1);
+  const values = date.split(separator);
+  const data = {};
+
+  for (let i = 0; i < parts.length; i++) {
+    data[parts[i]] = values[i];
+
+    if (parts[i] === YEAR_FORMAT_LONG) {
+      data[YEAR_FORMAT_SHORT] = values[i].substring(2);
+    }
+
+    if (parts[i] === YEAR_FORMAT_SHORT) {
+      data[YEAR_FORMAT_LONG]
+        = (+values[i] < CENTURY_THRESHOLD)
+          ? CENTURY_PREFIX_21 + values[i]
+          : CENTURY_PREFIX_20 + values[i];
+    }
+  }
+
+  return data;
+}
+
+/**
+ * @param {Object} data
+ * @param {string[]} format
+ *
+ * @returns {string}
+ */
+function arrangeDate(data, format) {
+  const parts = format.slice(0, -1);
+  const separator = format.slice(-1);
```

Comment:
You are using -1 so often if you made it in one function it could be separate const.

Reply:
Two times in one function is not so often. But it sure can be a constant, even if it is being used only once.

Reply:
I counted 4 times.
You also have same logic on 27, 28, lines.

Reply:
I know, that's why I wrote "two times in one function", and there's two functions, obviously :)

Reply:
Misunderstanding)